server/errorhandler.go

```diff
@@ -0,0 +1,47 @@
+package server
+
+import (
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/containous/traefik/middlewares"
+)
+
+// RecordingErrorHandler is an error handler, implementing the vulcand/oxy
+// error handler interface, which is recording network errors by setting the
+// request context value with the key middlewares.DefaultNetErrCtxKey to true.
+// This enables the retry middleware to only retry on actual network errors.
```

Comment:
I added a general comment that it will map to the proper status codes. I fear that being to explicit in the comment will bring up the danger, that the implementation and the comment will drift apart at some time. And when people read the comment, they probably also can read the code.